pkg/cloud_provider/file/file.go

```diff
@@ -860,39 +860,55 @@ func (manager *gcfsServiceManager) GetShare(ctx context.Context, obj *Share) (*S
 }
 
 func (manager *gcfsServiceManager) ListShares(ctx context.Context, filter *ListFilter) ([]*Share, error) {
-	uri := instanceURI(filter.Project, filter.Location, filter.InstanceName)
-	lCall := manager.multishareInstancesSharesService.List(uri).Context(ctx)
-	nextPageToken := "pageToken"
-	var shares []*Share
-
-	for nextPageToken != "" {
-		resp, err := lCall.Do()
+	instanceNames := make([]string, 0, 1)
+	if filter.InstanceName != "-" {
+		instanceNames = append(instanceNames, filter.InstanceName)
+	} else {
+		instances, err := manager.ListMultishareInstances(ctx, filter)
 		if err != nil {
-			klog.Errorf("list share error: %v", err)
 			return nil, err
 		}
+		for _, instance := range instances {
+			instanceNames = append(instanceNames, instance.Name)
+		}
+	}
 
-		for _, sobj := range resp.Shares {
-			project, location, instanceName, shareName, err := util.ParseShareURI(sobj.Name)
+	var shares []*Share
+
+	for _, instanceName := range instanceNames {
+		uri := instanceURI(filter.Project, filter.Location, instanceName)
+		lCall := manager.multishareInstancesSharesService.List(uri).Context(ctx)
+		nextPageToken := "pageToken"
+
+		for nextPageToken != "" {
+			resp, err := lCall.Do()
 			if err != nil {
-				klog.Errorf("Failed to parse share url :%s", sobj.Name)
-				continue
+				klog.Errorf("list share error: %v", err)
+				return nil, err
 			}
 
-			s := &Share{
-				Name: shareName,
-				Parent: &MultishareInstance{
-					Name:     instanceName,
-					Project:  project,
-					Location: location,
-				},
-				MountPointName: sobj.MountName,
-				CapacityBytes:  sobj.CapacityGb * util.Gb,
+			for _, sobj := range resp.Shares {
+				project, location, instanceName, shareName, err := util.ParseShareURI(sobj.Name)
+				if err != nil {
+					klog.Errorf("Failed to parse share url :%s", sobj.Name)
+					continue
+				}
+
+				s := &Share{
+					Name: shareName,
+					Parent: &MultishareInstance{
+						Name:     instanceName,
+						Project:  project,
+						Location: location,
+					},
+					MountPointName: sobj.MountName,
+					CapacityBytes:  sobj.CapacityGb * util.Gb,
```

Comment:
can you also capture labels and state in this object.
Also similar thing to be done in GetShare() https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/pkg/cloud_provider/file/file.go#L853

Reply:
done